frame/scored-pool/src/lib.rs

```diff
@@ -407,16 +407,20 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
 		new_members.sort();
 
 		let old_members = <Members<T, I>>::get();
+		// TODO: What to do in case of error?
+		let _ = T::MembershipChanged::ensure_can_change_members(&new_members, &old_members);
 		<Members<T, I>>::put(&new_members);
 
 		match notify {
 			ChangeReceiver::MembershipInitialized =>
 				T::MembershipInitialized::initialize_members(&new_members),
-			ChangeReceiver::MembershipChanged =>
-				T::MembershipChanged::set_members_sorted(
+			ChangeReceiver::MembershipChanged => {
+				// TODO: Is this safe?
```

Comment:
This is probably not safe as is because members here are sorted by account id, not score. So if there is truncation happening downstream, it will pick the wrong set of members.